img.js

```diff
@@ -122,8 +122,64 @@ function getValidWidths(originalWidth, widths = [], allowUpscale = false) {
   return filtered.sort((a, b) => a - b);
 }
 
+// Try to read memoized hashes from filesystem
+let imgHashCache = {};
+try {
+  imgHashCache = require(__dirname+"/.image-hash-cache.js");
+} catch(err) {
+  if(err.code !== 'MODULE_NOT_FOUND') {console.log(err);}
+}
+// Save hashes before exiting.
+process.on("exit", () => {
+  fs.writeFileSync(__dirname+"/.image-hash-cache.js", "module.exports = " + JSON.stringify(imgHashCache));
```

Comment:
I'm a bit skeptical about this line. Afaik, Windows uses back slashes, so maybe this won't work, or maybe Node will take care of it.

I couldn't find any info online, and I don't have a Windows machine with me to test, but I'll try to test it tomorrow.

Reply:
oh! I’m so sorry—I should have specified, I don’t think we need to memoize to the file system for this. I think that will cause more headache than it’s worth for right now. definitely keep the in-memory one though.

But for future reference, Eleventy already does have the convention of a `.cache/` folder and you can use `path.join` to normalize file paths if needed https://nodejs.org/dist/latest-v14.x/docs/api/path.html#path_path_join_paths

Reply:
After that’s in I think we’re good to merge it!

Reply:
No worries, it was super fun to do! :D

It felt a bit hacky though, so I separated it to a different commit. Glad we don't need it, I'll try to remove this commit.

Reply:
Yup, I reverted it.

> After that’s in I think we’re good to merge it!

Ooh I'm so excited!!